cmd/kube-proxy/app/server_windows.go

```diff
@@ -184,43 +150,18 @@ func newProxyServer(config *proxyconfigapi.KubeProxyConfiguration, master string
 		OOMScoreAdj:         config.OOMScoreAdj,
 		ConfigSyncPeriod:    config.ConfigSyncPeriod.Duration,
 		HealthzServer:       healthzServer,
-		UseEndpointSlices:   useEndpointSlices,
+		UseEndpointSlices:   true, // enable it by default.
 	}, nil
 }
 
 func getDualStackMode(networkname string, compatTester winkernel.StackCompatTester) bool {
 	return compatTester.DualStackCompatible(networkname)
 }
 
-func getProxyMode(proxyMode proxyconfigapi.ProxyMode, kcompat winkernel.KernelCompatTester) proxyconfigapi.ProxyMode {
-	if proxyMode == proxyconfigapi.ProxyModeKernelspace {
-		return tryWinKernelSpaceProxy(kcompat)
-	}
-	return proxyconfigapi.ProxyModeUserspace
-}
-
 func detectNumCPU() int {
 	return goruntime.NumCPU()
 }
 
-func tryWinKernelSpaceProxy(kcompat winkernel.KernelCompatTester) proxyconfigapi.ProxyMode {
-	// Check for Windows Kernel Version if we can support Kernel Space proxy
-	// Check for Windows Version
-
-	// guaranteed false on error, error only necessary for debugging
-	useWinKernelProxy, err := winkernel.CanUseWinKernelProxier(kcompat)
-	if err != nil {
-		klog.ErrorS(err, "Can't determine whether to use windows kernel proxy, using userspace proxier")
-		return proxyconfigapi.ProxyModeUserspace
-	}
-	if useWinKernelProxy {
-		return proxyconfigapi.ProxyModeKernelspace
-	}
-	// Fallback.
-	klog.V(1).InfoS("Can't use winkernel proxy, using userspace proxier")
```

Comment:
I don't know the situation on Windows, but if there are systems that can run Kubernetes that can't run the kernel proxier, then we might want to keep the explicit check and have it fail with a clear error, rather than just crashing with an EINVAL or whatever later on.